src/suggestion-box.ts

```diff
@@ -0,0 +1,482 @@
+            }
+        }
+        if (this.index >= 0 && this.index < resultLength) {
+            const selectedItem = this.private_items[this.index];
+
+            if (selectedItem) {
+                this.private_items[this.prevIndex]?.classList.remove(
+                    `${this.activecls}`
+                );
+                selectedItem.classList.add(`${this.activecls}`); // here it represents the items in the result list so we need to handle this
+            }
+        } else if (this.index == resultLength && this.has_more) {
+            this.private_items[this.index].classList.add(`${this.activecls}`);
+            this.private_items[this.prevIndex]?.classList.remove(
+                `${this.activecls}`
+            ); // here it represents loadmore, index and prevIndex are different so we need this lines
+        }
+        this.prevIndex = this.index;
+    }
+
+    private _itemClickedEvent(event: Event) {
+        let indextarget = this._indexOfTarget(event);
+        if (this.result[indextarget]) {
+            this._setInputEvent(indextarget);
+        } else if (indextarget === this.result.length) {
+            this._loadmoreEntry(event);
+        }
+        this._scrollIntoView();
+    }
+
+    private _setInputEvent(index: number) {
+        let selectedValue: any;
+        if (typeof this.result[index] === "string") {
+            selectedValue = this.result[index];
+        } else {
+            selectedValue = this.result[index][this.listkey];
+        }
+        if (selectedValue) {
+            (this.input_el as HTMLInputElement).value = selectedValue;
+            this.show_container = false;
+        }
+    }
+
+    private _highlight(text: string, match_str: string) {
+        if (!match_str) return text;
+        const regex = new RegExp(`(${match_str.split(" ").join("|")})`, "gi");
+        return text.replace(regex, (match) => `<strong>${match}</strong>`);
+    }
+
+    _startsWithHighlight(name: any) {
+        const input_val = (this.input_el as HTMLInputElement)?.value;
+        if (!input_val) {
+            return name;
+        }
+
+        const escaped_input_val = input_val.replace(
+            /[-\/\\^$*+?.()|[\]{}]/g,
+            "\\$&"
+        );
+        const re = new RegExp(
+            "(" + escaped_input_val.split(" ").join("|") + ")",
+            "gi"
+        );
+        return name.replace(re, `<strong class="sgn-highlight">$1</strong>`);
+    }
+    setSuggestion(
+        suggestions: Array<{ [key: string]: string } | string>,
+        has_more: boolean
+    ) {
+        this.result = [...suggestions];
+        this.index = -1;
+        (this.input_el as HTMLInputElement)?.value
+            ? (this.show_empty = false)
+            : (this.show_empty = true);
+        this.has_more = has_more;
+        if (this.private_items?.length) {
+            for (let el of this.private_items) el.remove();
+        }
+        this.filtered_list = suggestions;
+        this.show_container = true;
```

Comment:
Move the "No matching results" code from the inputEvent to this function. This is the best place to do that, as it will accomodate both the predefined list case, as well as the other case